internal/controller/networking/ingress_controller.go

```diff
@@ -187,13 +219,156 @@ func (r *IngressReconciler) gatherApiCheckData(ingress *networkingv1.Ingress) (a
+				if ingress.Annotations[annotationPath] == "" {
+					if rulePath.Path == "" {
+						paths = append(paths, "/")
+					} else {
+						paths = append(paths, rulePath.Path)
+					}
+				} else {
+					paths = append(paths, ingress.Annotations[annotationPath])
+				}
+			}
+		}
+
+		for _, path := range paths {
+			// Replace path /
+			path = strings.TrimPrefix(path, "/")
+
+			// Set apiCheck Name
+			checkName := fmt.Sprintf("%s-%s-%s", ingress.Name, host, path)
+			checkName = strings.Replace(checkName, "/", "", -1)
+			checkName = strings.Replace(checkName, ".", "", -1)
+			checkName = strings.Trim(checkName, "-")
+
+			// Set endpoint
+			endpoint := fmt.Sprintf("https://%s/%s", host, path)
+
+			// Construct ApiCheck Spec
+			apiCheckSpec := &checklyv1alpha1.ApiCheckSpec{
+				Endpoint: endpoint,
+				Group:    group,
+				Success:  success,
+				Muted:    muted,
+			}
+
+			newApiCheck := &checklyv1alpha1.ApiCheck{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      checkName,
+					Namespace: ingress.Namespace,
+					OwnerReferences: []metav1.OwnerReference{
+						*metav1.NewControllerRef(ingress, networkingv1.SchemeGroupVersion.WithKind("ingress")),
+					},
+					Labels: labels,
+				},
+				Spec: *apiCheckSpec,
+			}
+
+			apiChecks = append(apiChecks, newApiCheck)
+		}
 	}
 
 	// Last return
 	return
 }
+
+func (r *IngressReconciler) compareApiChecks(ctx context.Context, ingress *networkingv1.Ingress, ingressApiChecks []*checklyv1alpha1.ApiCheck) (newApiChecks []*checklyv1alpha1.ApiCheck, deleteApiChecks []*checklyv1alpha1.ApiCheck, updateApiChecks []*checklyv1alpha1.ApiCheck, err error) {
+
+	logger := log.FromContext(ctx)
+
+	var existingApiChecks checklyv1alpha1.ApiCheckList
+	labels := make(map[string]*string)
+	labels["ingress-controller"] = &ingress.Name
+	err = r.List(ctx, &existingApiChecks, client.InNamespace(ingress.Namespace), client.MatchingLabels{"ingress-controller": ingress.Name})
+	if err != nil {
+		return
+	}
+
+	existingApiChecksMap := make(map[string]checklyv1alpha1.ApiCheck)
+	for _, existingApiCheck := range existingApiChecks.Items {
+		existingApiChecksMap[existingApiCheck.Name] = existingApiCheck
+	}
+
+	newApiChecksMap := make(map[string]*checklyv1alpha1.ApiCheck)
+	for _, ingressApiCheck := range ingressApiChecks {
+		newApiChecksMap[ingressApiCheck.Name] = ingressApiCheck
+	}
+
+	// Compare items
+	for _, existingApiCheck := range existingApiChecksMap {
+		_, exists := newApiChecksMap[existingApiCheck.Name]
+		if exists {
+			if existingApiCheck.Spec == newApiChecksMap[existingApiCheck.Name].Spec {
+				logger.Info("ApiCheck data is identical, no need for update", "ApiCheck Name", existingApiCheck.Name)
+			} else {
+				logger.Info("ApiCheck data is not identical, update needed", "ApiCheck Name", existingApiCheck.Name, "old spec", existingApiCheck.Spec, "new spec", newApiChecksMap[existingApiCheck.Name].Spec)
+				updateApiChecks = append(updateApiChecks, newApiChecksMap[existingApiCheck.Name])
```

Comment:
I think it would be better to just use the map value you've retrieved instead of constantly accessing the value through the map by name, which is prone to mistakes.
 
```suggestion
		newApiCheck, exists := newApiChecksMap[existingApiCheck.Name]
		if exists {
			if existingApiCheck.Spec == newApiCheck.Spec {
				logger.Info("ApiCheck data is identical, no need for update", "ApiCheck Name", existingApiCheck.Name)
			} else {
				logger.Info("ApiCheck data is not identical, update needed", "ApiCheck Name", existingApiCheck.Name, "old spec", existingApiCheck.Spec, "new spec", newApiCheck.Spec)
				updateApiChecks = append(updateApiChecks, newApiCheck)
```

Reply:
Fixed in latest commit.